CMake/External_OpenCV.cmake

```diff
@@ -41,6 +61,20 @@ else()
   list(APPEND OpenCV_EXTRA_BUILD_FLAGS -DWITH_FFMPEG=OFF)
 endif()
 
+# Set VTK dependency if we're locally building it
+if(fletch_ENABLE_OpenCV_VTK)
+  message(STATUS "OpenCV depending on fletch VTK")
+  if (OpenCV_version VERSION_LESS 3.2.0 AND VTK_version VERSION_GREATER_EQUAL 7.0)
+    message(FATAL_ERRROR "OpenCV versions before 3.2 can only handle pre 7.0 VTK versions")
+  endif()
+  list(APPEND OpenCV_DEPENDS VTK)
+  list(APPEND OpenCV_EXTRA_BUILD_FLAGS -DWITH_VTK:BOOL=TRUE
+    -DVTK_DIR:PATH=${fletch_BUILD_INSTALL_PREFIX}/lib/cmake/vtk-${VTK_version}
```

Comment:
In the current state of the PR turning on VTK will actually break because External_OpenCV is included before External_VTK. Originally I re-arranged the order in fletch-tarballs, but that introduced a lot of merge conflicts in my upstream so I reverted for the time being.

I'm currently leaning towards the idea of removing the option to enable VTK support for OpenCV, but I think its still a good idea to keep this logic here in case we do need to enable it in the future. That being said, I'll change to use the VTK_DIR. We'd need to make that ordering change anyway.